### PR TITLE
Fix Compilation for 1.20.1

### PR DESCRIPTION
**Pull request by TheRealRyGuy**
# Description
Fixes compilation to 1.20.1 (#792) and supersedes #793 
(was just hoping to get this pr across the finish line for 1.20.1 support cause i need it)

- This does not add FeatureFlag, Folia API, or transient attribute modifiers, just throws `UnimplementedOperationException` for compilation
- Did not add test for implemented `EndermanMock` methods, as they are random in nature and pulled directly from NMS. They could use return value tests though I guess?
- Figured methods not pulled directly from NMS or self explanatory could just have functionality implemented at a later date, as end goal was to just get this out as soon as possible
  - Jitpack's 1.20 [failed to build](https://jitpack.io/com/github/MockBukkit/MockBukkit/v1.20-v3.3.1-ga54c5e8-1/build.log), as did [latest 1.20 branch publish](https://github.com/MockBukkit/MockBukkit/actions/runs/5316249643)
# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -6,7 +6,7 @@ plugins {
 	id("maven-publish")
 	id("signing")
 	id("net.kyori.blossom") version "1.3.1"
-	id("io.github.gradle-nexus.publish-plugin") version "1.3.0"
+	id("io.github.gradle-nexus.publish-plugin") version "2.0.0-rc-1"
```
Comment (Insprill) [inline build.gradle.kts]: Please revert this change. Modifying the build file will require a review from seeseemelk before being merged. We also shouldn't switch to pre-release versions before they're ready.

File: build.gradle.kts [hunk 2]
```diff
@@ -29,7 +29,7 @@ dependencies {
 	// General utilities for the project
 	implementation("net.kyori:adventure-platform-bungeecord:4.3.0")
 	implementation("org.jetbrains:annotations:24.0.1")
-	implementation("net.bytebuddy:byte-buddy:1.14.4")
+	implementation("net.bytebuddy:byte-buddy:1.14.5")
```
Comment (Insprill) [inline build.gradle.kts]: Same goes for this. There's already a renovate PR open for it.

**Review by Insprill — CHANGES_REQUESTED** [review]
There's also some code style fixes. Please run IntelliJ's formatter to fix them.

**Comment by TheRealRyGuy** [comment]
> There's also some code style fixes. Please run IntelliJ's formatter to fix them.

`Reformat Code` on IntelliJ modifies 300+ files, primarily json resources, but plenty of files I didn't modify - not sure if that's what you meant. Reverted dependency change though, didn't know about the review requirements 
- `Enable EditorConfig support` is enabled, so I'm not 100% sure

**Comment by Insprill** [comment]
I just meant to run it on the files/sections of code you modified

**Comment by TheRealRyGuy** [comment]
Done, apologies for this misunderstanding - everything builds properly

**Comment by Insprill** [comment]
Please revert that commit and rerun it without reorganize code checked—apologies for the back-and-forth.

**Comment by TheRealRyGuy** [comment]
reverted and reformatted through a force push to concise commit history, am gonna fall asleep so feel free to override anything if you don't want to wait until tomorrow, it all looks good though

**Review by Insprill — APPROVED** [review]
LGTM

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java [hunk 3]
```diff
@@ -1059,6 +1059,14 @@ void testPotionEffects()
...
 	}
 
+	@Test
+	void clearPotionEffects()
+	{
+		PotionEffect effect = new PotionEffect(PotionEffectType.CONFUSION, 5, 1);
+		player.addPotionEffect(effect);
+		assertTrue(player.clearActivePotionEffects());
+	}
+
```
Comment (thelooter) [inline src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java]: Is there a reason you are testing this on PlayerMockTest and not LivingEntityMockTest

File: src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java [hunk 4]
```diff
@@ -1384,6 +1385,14 @@ public void sendSignChange(@NotNull Location loc, @Nullable String @Nullable []
 		}
 	}
 
+	@Override
+	public void sendBlockUpdate(@NotNull Location loc, @NotNull TileState tileState) throws IllegalArgumentException
+	{
+		Preconditions.checkNotNull(loc);
+		Preconditions.checkNotNull(tileState);
+		//Pretend we sent block update
+	}
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java:1394]: Can you please add a Test checking that both the Preconditions throw an exceptioon when provided with a null Value and a Test that checks that no exception is thrown when both values are non-null

File: src/main/java/be/seeseemelk/mockbukkit/entity/EndermanMock.java [hunk 5]
```diff
@@ -79,6 +84,36 @@ public void setCarriedBlock(@Nullable BlockData blockData)
...
+					(this.getLocation().y() + 1.45) - entity.getLocation().y(), this.getLocation().z() - entity.getLocation().z());
+
+			vector = vector.normalize();
+			double x = this.getLocation().x() + (random.nextDouble() - 0.5D) * 8.0D - vector.getX() * 16.0D;
+			double y = this.getLocation().y() + (double) (random.nextInt(16) - 8) - vector.getY() * 16.0D;
+			double z = this.getLocation().z() + (random.nextDouble() - 0.5D) * 8.0D - vector.getZ() * 16.0D;
+			return this.teleport(new Location(getWorld(), x, y, z));
+		}
+		return false;
+	}
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/entity/EndermanMock.java:115]: Can you also please add Tests for this? It should be possible to just check if the Location after the method call is different from the one beforehand.

**Review by thelooter — CHANGES_REQUESTED** [review]
A few little things

File: src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java (hunk 3 above)
Comment (TheRealRyGuy) [reply src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java]: PlayerMockTest has all potion effect tests that should probably be in LivingEntityMockTest - should all of these be refactored?

Comment (thelooter) [reply src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java]: yes

File: src/main/java/be/seeseemelk/mockbukkit/entity/EndermanMock.java (hunk 5 above)
Comment (thelooter) [reply src/main/java/be/seeseemelk/mockbukkit/entity/EndermanMock.java:115]: Maybe also check that the enderman is closer to the player than before

Comment (TheRealRyGuy) [reply src/main/java/be/seeseemelk/mockbukkit/entity/EndermanMock.java:115]: not crazy about testing distance, it can be arbitrary 
example: 
```java
@Test
void testRandomTeleportToEntity()
{
        Location loc = endermanMock.getLocation();
	PlayerMock player = serverMock.addPlayer();
	WorldMock mock = serverMock.addSimpleWorld("world");
	player.teleport(new Location(mock, 20, 20, 20));
	endermanMock.teleport(new Location(mock, 1, 1, 1));
	double distance = endermanMock.getLocation().distanceSquared(player.getLocation());
	assertTrue(endermanMock.teleportTowards(player));
	assertNotEquals(loc, endermanMock.getLocation());
	assertTrue(distance > endermanMock.getLocation().distanceSquared(player.getLocation()));
}
```
Distance test is not concrete- if the player is teleported to 5, 5, 5, the test fails, but at 20, 20, 20, the test passes

**Comment by TheRealRyGuy** [comment]
Condensed commit history and refactored potion tests to living entity, see endermanmock test comment. Builds and tests properly
